source/particle/world.cc

```diff
@@ -832,11 +840,18 @@ namespace aspect
               for (typename std::map<double, typename parallel::distributed::Triangulation<dim>::active_cell_iterator>::const_iterator neighbor_cell = neighbor_cells.begin();
                    neighbor_cell != neighbor_cells.end(); ++neighbor_cell)
                 {
-                  if (particle_is_in_cell(recv_particle,neighbor_cell->second) && neighbor_cell->second->is_locally_owned())
+                  try
                     {
-                      cell = neighbor_cell->second;
-                      break;
+                      const Point<dim> p_unit = this->get_mapping().transform_real_to_unit_cell(neighbor_cell->second, recv_particle.get_location());
+                      if (GeometryInfo<dim>::is_inside_unit_cell(p_unit))
+                        {
+                          cell = neighbor_cell->second;
+                          recv_particle.set_reference_location(p_unit);
+                          break;
+                        }
                     }
```

Comment:
This control flow is a bit hard to read. Can you just add a line at the end of the `try` block that says `// fall through, check next cell` or similar